src/cpp/type-conversion.cpp

```diff
@@ -298,6 +298,16 @@ namespace TypeConversion {
         // Handle GstStructure (like stats property)
         const GstStructure *structure = GST_STRUCTURE(boxed_value);
         return gst_structure_to_js(env, structure);
+      } else if (boxed_value && G_VALUE_TYPE(gvalue) == G_TYPE_VALUE_ARRAY) {
+        // Only cast to GValueArray if we're certain it's actually a GValueArray
+        GValueArray *arr = static_cast<GValueArray *>(boxed_value);
+        if (arr && arr->n_values > 0) { // sanity check with null check
+          Napi::Array jsArr = Napi::Array::New(env, arr->n_values);
+          for (guint i = 0; i < arr->n_values; i++) {
+            jsArr.Set(i, gvalue_to_js(env, &arr->values[i]));
+          }
+          return jsArr;
```

Comment:
Should be snake case here:

```suggestion
          Napi::Array js_arr = Napi::Array::New(env, arr->n_values);
          for (guint i = 0; i < arr->n_values; i++) {
            js_arr.Set(i, gvalue_to_js(env, &arr->values[i]));
          }
          return js_arr;
```